packages/multichain-account-service/src/providers/BaseBip44AccountProvider.ts

```diff
@@ -53,47 +61,62 @@ export type Bip44AccountProvider<
   resyncAccounts(accounts: Bip44Account<InternalAccount>[]): Promise<void>;
 };
 
-export abstract class BaseBip44AccountProvider implements Bip44AccountProvider {
+export abstract class BaseBip44AccountProvider<
+  Account extends Bip44Account<KeyringAccount> = Bip44Account<KeyringAccount>,
+> implements Bip44AccountProvider
+{
   protected readonly messenger: MultichainAccountServiceMessenger;
 
+  accountsList: Set<Bip44Account<KeyringAccount>['id']> = new Set();
```

Comment:
This should be `protected` probably?

Also, I'd just with `accounts`, that sounds more concise to me. WDYT?

Reply:
Yeah true, changing it to `protected`. I decided to have it as `accountsList` because we already have `accounts` in the mocked provider and so it conflicts.